storage_service/storage_service/settings/components/s3.py

```diff
@@ -0,0 +1,13 @@
+"""Configure S3
+
+From here we can configure aspects of S3 in the Storage Service.
+"""
+from os import environ
+
+# Read and connect timeouts for S3. Ideally these will match the
+# defaults recommended by your S3 implementation.
+S3_TIMEOUTS = 900
+try:
+    S3_TIMEOUTS = int(environ.get("SS_S3_TIMEOUTS", S3_TIMEOUTS))
+except ValueError:
+    pass
```

Comment:
When it comes to config the preferred pattern is fail-fast, i.e. report errors and exit, so the administrator receives feedback right away. We inherit that model from Django but it's pretty common in other systems, e.g. Django raises `django.core.exceptions.ImproperlyConfigured` (we do too from `get_env_variable`).
An alternative is to leverage Django's [system checks framework](https://docs.djangoproject.com/en/1.8/topics/checks/), but I think that's a solution we haven't explored yet.

Reply:
When you say the preferred pattern do you have a link to anything in our documentation that describes this? Or do we just work it out? It might seem like hard work @sevein but 3 years later these CRs are still painful-as because we haven't any of this stuff written down when it comes up. If you take anything away nit-picking the undocumented assumptions in our work is just not a good way to create the harmony needed to work on a code-base at scale. Believe it or not, I'm not actually trying to get this wrong the first time around.

The change has been made to raise improperly configured with a helpful message.

Reply:
No sorry, we don't have docs for that. I can refer to common software design patterns I see out there, say for example in Redis there is a `protected-mode` config directive that expects a boolean, and if you pass something else then you see something like the following:

```
*** FATAL CONFIG FILE ERROR (Redis 6.0.10) ***
Reading the configuration file, at line 87
>>> 'protected-mode 1'
argument must be 'yes' or 'no'
```

> If you take anything away nit-picking the undocumented assumptions in our work is just not a good way to create the harmony needed to work on a code-base at scale.

Noted, thanks.